homeassistant/components/tado/climate.py

```diff
@@ -496,6 +506,11 @@ def set_hvac_mode(self, hvac_mode: HVACMode) -> None:
         """Set new target hvac mode."""
         self._control_hvac(hvac_mode=HA_TO_TADO_HVAC_MODE_MAP[hvac_mode])
 
+    def set_heating_circuit(self, heater:str=None):
+        if heater:
+            heater = 1
```

Comment:
so you basically select a water heater that is connected to tado (and exposed through tado as water heater entity), correct? Why is this set to `1` regardless of the input?

Reply:
Because currently there is no way to fetch heater id from heater entity_id, as I said earlier It just can assign or unassign the heater from zone. the `heater = 1` means it ll use the heater with id 1.

Reply:
which is the first one in tado (or in tado in the zone), or what is the heater with id 1? Wouldn't the service then rather be "set_default_heater" with a boolean input? Otherwise it's going to be absolutely confusing for the end user. Also, the zone used is the zone of the climate entity?

Reply:
I was considering the use of a boolean for the function, as you mentioned, but it's not future-proof enough. In a typical heating circuit, there is usually just one source of heat. However, there are setups with multiple sources or circuits, which complicates things. Currently, I can only program for one water heater. As a result, it's difficult to test what kind of results I would get from the Tado API with more than 1 water heaters.

Reply:
> Also, the zone used is the zone of the climate entity?

Every zone has own climate entity, so I used in code just target of climate entity to select for witch zone user want to set heating circuit

Reply:
My tado system doesn't give me the optino at all (probably because I have only thermostats). I would still request to don't set something that isn't use. Maybe make two services instead enable_/disable_default(?)_zone_heater without any input parameters. Otherwise nobody will know/understand how this works